interoperator/controllers/schedulers/sfserviceinstanceupdater/sfserviceinstanceupdater_controller.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/go-logr/logr"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// SFServiceInstanceUpdater reconciles a SFServiceInstance object
+type SFServiceInstanceUpdater struct {
+	client.Client
+	Log             logr.Logger
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile schedules the SFServiceInstance to one SFCluster and sets the ClusterID in
+// SFServiceInstance.Spec.ClusterID. It chooses the destination cluster based on clusterSelector
+// template provided in the plan.
+func (r *SFServiceInstanceUpdater) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfserviceinstance-updater", req.NamespacedName)
+
+	plan := &osbv1alpha1.SFPlan{}
+	err := r.Get(ctx, req.NamespacedName, plan)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if plan.Spec.AutoUpdatable == false {
+		arrBytes := []byte{}
+		jsonBytes, _ := json.Marshal(plan.Spec)
+		arrBytes = append(arrBytes, jsonBytes...)
+		hash := md5.Sum(arrBytes)
+		currentSpecHash := base64.StdEncoding.EncodeToString(hash[:])
+		if plan.Status.SpecHash == "" {
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				err1 := r.Get(ctx, req.NamespacedName, plan)
+				if err1 != nil {
+					return err1
+				}
+				plan.Status.SpecHash = currentSpecHash
+				return r.Status().Update(ctx, plan)
+			})
+			if err != nil {
+				log.Error(err, "Error occured while updating the sfplan spec-hash", "cluster-name", plan.GetName(), "current-hash", currentSpecHash)
+				return ctrl.Result{}, err
+			}
+		} else if plan.Status.SpecHash != currentSpecHash {
+			var instances osbv1alpha1.SFServiceInstanceList
+			err = r.List(context.Background(), &instances, client.MatchingField("spec.planId", plan.ObjectMeta.Name))
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			for _, instance := range instances.Items {
+				log.Info("Update required for : Instance with plan ID", "plan ID", plan.ObjectMeta.Name, "SFServiceInstance ID", instance.Name)
+				//Set state as update for instances
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					err1 := r.Get(ctx, types.NamespacedName{
+						Name:      instance.GetName(),
+						Namespace: instance.GetNamespace(),
+					}, &instance)
+					if err1 != nil {
+						return err1
+					}
+					instance.Status.State = "update"
+					return r.Update(ctx, &instance)
+				})
+				if err != nil {
+					log.Error(err, "Error occured while updating the instance", "instance-name", instance.GetName())
+					return ctrl.Result{}, err
+				}
+			}
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Should we do the hash calculation also within the retry?

Reply:
👍

Reply:
done. Can you please check if it looks good now ?